src/main/java/org/broadinstitute/dsm/jobs/UPSTrackingJob.java

```diff
@@ -21,60 +22,230 @@
+                            if (!isReturn) {
+                                updateTrackingInfo(statusType, oldType, statusDescription, trackingId, date, SQL_UPDATE_UPS_TRACKING_STATUS, isReturn, kit);
+                            }
+                            else {
+                                updateTrackingInfo(statusType, oldType, statusDescription, trackingId, date, SQL_UPDATE_UPS_RETURN_STATUS, isReturn, kit);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    private static void updateTrackingInfo(String statusType, String oldType, String statusDescription, String trackingId, String date, String query, boolean isReturn, DdpKit kit) {
+        String upsUpdate = statusType + " " + statusDescription;
+        SimpleResult result = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setString(1, upsUpdate);
+                stmt.setString(2, date);
+                stmt.setString(3, trackingId);
+                int r = stmt.executeUpdate();
+                if (r != 2) {//number of subkits
+                    throw new RuntimeException("Update query for UPS tracking updated " + r + " rows!");
+                }
+            }
+            catch (Exception e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+        if (result.resultException != null) {
+            throw new RuntimeException(result.resultException);
+        }
+        else {
+            if (!isReturn) {
+                //if delivered notif pepper
+                if (DELIVERY.equals(statusType) && !(DELIVERY.equals(oldType))) {
+                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] { DELIVERED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
+                    if (kitDDPNotification != null) {
+                        logger.info("Triggering DDP for delivered kit with external order number: " + kit.getExternalOrderNumber());
+                        EventUtil.triggerDDP(kitDDPNotification);
+
+                    }
+                    else {
+                        logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
+                    }
+                }
 
             }
-            catch (IOException e) {
-                throw new RuntimeException("couldn't get response from ups tracking ", e);
+            else {
+                //if picked up place order
+                if (!(PICKUP.equals(oldType)) && !(IN_TRANSIT.equals(oldType)) && (PICKUP.equals(statusType) || IN_TRANSIT.equals(statusType))) {
```

Comment:
Maybe factor out L207 to a method like `isBeingReturned(oldType, newType)`?

Reply:
Yes I think we should decide on what beingReturned means for us

Reply:
I think L207 is a good start--let's move ahead with this and expand as we see other patterns.  Are there UPS mail innovation docs that describe each status somewhere?

Reply:
no they are very summarized, and I can see now in logs that they are not really totally correct